### PR TITLE
feat(relayer/txmgr): additional txmgr metrics

### PR DESCRIPTION
**Pull request by Lazar955**
adds more metrics for txmgr 

task: none

### CODE REVIEW

**Review by DanFlannel — APPROVED** [review]
LGTM

File: relayer/txmgr/metrics.go [hunk 1]
```diff
@@ -13,4 +13,25 @@ var (
...
+		Name:      "tx_fee_gwei_total",
+		Help:      "Sum of fees spent for all transactions in GWEI",
+	}, []string{"chain"})
+
+	txConfirmationLatency = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: "relayer",
+		Subsystem: "txmgr",
+		Name:      "tx_confirmation_latency",
+		Help:      "Latency between transaction submission and confirmation in seconds",
+	}, []string{"chain"})
```
Comment (pavelnikolov) [inline relayer/txmgr/metrics.go:36]: I'm not sure about the nature of this latency, but I'm used to always use histograms for durations. This way we can visualize/calculate the 99th percentile latency, and remove outliers from the graphs (better for alerting). I'm not sure if we care about that. 
In addition to that with a histogram we can display a heatmap graph.
But I'm not sure if that's applicable here. I'm happy to show an example.

**Review by pavelnikolov — APPROVED** [review]

File: relayer/txmgr/txmgr.go [hunk 2]
```diff
@@ -359,6 +360,13 @@ func (m *SimpleTxManager) sendTx(ctx context.Context, tx *types.Transaction) (*t
 			return nil, errors.Wrap(ctx.Err(), "context canceled")
 
 		case receipt := <-receiptChan:
+			if receipt.EffectiveGasPrice == nil {
+				return receipt, nil
+			}
+			fee := float64(receipt.EffectiveGasPrice.Uint64() * receipt.GasUsed / params.GWei)
```
Comment (corverroos) [inline relayer/txmgr/txmgr.go:366]: lets rather track effective gas price as a gauge.

File: relayer/txmgr/txmgr.go [hunk 3]
```diff
@@ -437,6 +446,7 @@ func (m *SimpleTxManager) waitForTx(ctx context.Context, tx *types.Transaction,
 	}
 	select {
 	case receiptChan <- receipt:
+		txConfirmationLatency.WithLabelValues(m.chainName).Set(time.Since(t).Seconds())
```
Comment (corverroos) [inline relayer/txmgr/txmgr.go:449]: put this above the `select`.

File: relayer/txmgr/metrics.go (hunk 1 above)
Comment (corverroos) [reply relayer/txmgr/metrics.go:36]: So yes, normally latencies should be a histogram. Especially if the individual measurement are indiepedent of each other. Your HTTP request and my HTTP request is concurrent and unrelated.

But in blocks chain, we do things sequentially. One after the other. In this case, txs are sent and then confirmed one after the other. So the lag/latency is correlated. The second cannot confirm until the first is confirmed.

So a gauge is fine and simpler in this case.

File: relayer/txmgr/metrics.go [hunk 4]
```diff
@@ -13,4 +13,25 @@ var (
 		Name:      "resend_total",
 		Help:      "The total number of transaction resends to a destination chain",
 	}, []string{"chain"})
+
+	txL1GasFee = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: "relayer",
+		Subsystem: "txmgr",
+		Name:      "tx_fee_gwei",
+		Help:      "L1 gas fee for transactions in GWEI",
```
Comment (corverroos) [inline relayer/txmgr/metrics.go:21]: I think this should actually be a histogram, since individual submissions's fee are uncorrelated. And I would rather just track `GasUsed`.

**Review by corverroos — APPROVED** [review]
TL;DR:

Lets rather track: 
   - histo:`gas_used` (histo because they are uncorrelated)
   - gauge:`effective_gas_price` (gauge because they are correlated)

File: relayer/txmgr/metrics.go (hunk 1 above)
Comment (pavelnikolov) [reply relayer/txmgr/metrics.go:36]: Understood and I agree about that - the current value is enough to monitor the system health as a whole. But still in alerts we can track 95th or 99th percentile latency and ignore sporadical outliers.
